### ISSUE TITLE
Document Preprocessor for PDF Documents

### ISSUE THREAD

**Issue by Allen8838** [HazyResearch/fonduer#91]
Hello,

I was wondering if there is a document preprocessor for PDF documents. I tried using the DocPreprocessor command but it looks like the parse_file function has not been built out yet. In the tutorial notebooks, it looks like both html files and pdf files are processed together, but I was wondering how I would go about parsing sentences for only pdf files. If there is some documentation I should be referring to, please let me know. Thanks.

**Comment by lukehsiao** [comment]
Hi @Allen8838, 

>I tried using the DocPreprocessor command but it looks like the parse_file function has not been built out yet.

As you've noticed, DocPreprocessor is just the base class that the other preprocessors inherit from and isn't meant to be used bare. 

>I was wondering if there is a document preprocessor for PDF documents. [...] In the tutorial notebooks, it looks like both html files and pdf files are processed together, but I was wondering how I would go about parsing sentences for only pdf files.

**TLDR: For now, you'll need to get an HTML representation of your PDF documents using a third party tool like Adobe Acrobat. We're working on making this a more seamless process in the future.**

Your observation here is also correct. To give you a bit more context, the way that Fonduer is currently built expects that there is an HTML representation of all PDF input documents. In our experiments, we previously had used Adobe Acrobat to convert PDFs to HTML, with reasonably good results. Others have also used tools from ABBYY to get the HTML representation to go along with the PDF files. 

An HTML representation allows us to extraction structural/tabular information more easily than trying to do so directly from the PDF and also helps us deal with some of the nuanced corner cases involved in parsing PDF files.

However, Adobe can be quite a bottleneck. Acrobat Pro has a memory leak when converting documents, and performs quite slowly, making converting a large PDF corpus to HTML quite a painful process. [pdftotree](https://github.com/HazyResearch/pdftotree) is our current effort to eliminate the need for proprietary tools like Adobe by constructing a document structure ourselves from the raw PDF. Ultimately, [pdftotree](https://github.com/HazyResearch/pdftotree) will be used in Fonduer to preprocess PDFs when necessary so that users do not need to supply their own HTML representations.

**Comment by Allen8838** [comment]
Thank you Luke.
